### ISSUE TITLE
alias_method_chain is deprecated

### ISSUE THREAD

**Issue by 2called-chaos** [brentd/xray-rails#81]
As the title already tells, xray-rails is using `alias_method_chain` to patch `ActionView::Template#render` but it produces deprecation warnings.

[The fix](https://github.com/2called-chaos/xray-rails/commit/4d986e37ea8c88569a90e43cfa1f015db9d197c8) is quite easy but I don't know if you want this as a PR since I just put the module there and it removes compatibility with Ruby <2.0

Let me know what you think :)

---

```
DEPRECATION WARNING: alias_method_chain is deprecated. Please, use Module#prepend instead. From module, you can access the original method using super. (called from <top (required)> at /Volumes/code/_bmonkeys/tomatic/config/environment.rb:5)
	/Users/chaos/.gem/ruby/2.3.0/gems/xray-rails-0.2.0/lib/xray/engine.rb:40:in `block (2 levels) in <class:Engine>'
	/Users/chaos/.gem/ruby/2.3.0/gems/xray-rails-0.2.0/lib/xray/engine.rb:21:in `class_eval'
	/Users/chaos/.gem/ruby/2.3.0/gems/xray-rails-0.2.0/lib/xray/engine.rb:21:in `block in <class:Engine>'
```

**Comment by mattbrictson** [comment]
Thanks for starting the discussion on this!

Even though Ruby 1.9 is ancient history at this point, I would like to maintain compatibility if possible. Is there a way we can support both techniques by checking `Module.respond_to?(:prepend)` first?

**Comment by 2called-chaos** [comment]
What do you think of this? It's a little bit messy due to the alias_method within the new patch module but this would reduce duplication to a minimum I suppose...

https://github.com/2called-chaos/xray-rails/commit/fd52f4ec32f22c04c763a3bb3a86ba506f1783b2

Note: I only tested both ways with Ruby 2.3

**Comment by mattbrictson** [comment]
Hmm, looks a bit messy but I think that is unavoidable. My only suggestion would be to move the `alias_method :render_with_xray, :render` line down to just before you call `alias_method_chain`.

Could you open a PR? Thanks! Please include an entry in the changelog.

**Comment by 2called-chaos** [comment]
You mean in between `include` and `alias_method_chain`? I think this doesn't work. I initially tried

```
alias_method :render_without_xray, :render
instance_eval { undef :render } # optionally but after undef it doesn't include anyway
include Xray::Engine::ActionViewPatch
# at this point I can't get the patched method unless I alias_method it in the module
```

**Comment by mattbrictson** [comment]
I think this works:

```ruby
include Xray::Engine::ActionViewPatch
Xray::Engine::ActionViewPatch.send :alias_method, :render_with_xray, :render
alias_method_chain :render, :xray
```

**Comment by 2called-chaos** [comment]
I think we did it :) I found it a bit odd that you can alias_method _after_ the include but I confirmed it to work on 1.9.3 as well.

**Comment by mattbrictson** [comment]
Fixed via #83
